CMakeLists.txt

```diff
@@ -62,8 +62,17 @@ option(ENABLE_AIRTUNES    "Enable AirTunes support?" ON)
 option(ENABLE_OPTICAL     "Enable optical support?" ON)
 option(ENABLE_PYTHON      "Enable python support?" ON)
 option(ENABLE_TESTING     "Enable testing support?" ON)
+option(ENABLE_LIBTORRENT  "Enable libtorrent support?" ON)
 
 # Internal Depends - supported on all platforms
+option(ENABLE_INTERNAL_BOOST "Enable internal boost?" ON) # TODO: Change to OFF
+option(ENABLE_INTERNAL_LIBDATACHANNEL "Enable internal libdatachannel?" ON) # TODO: Change to OFF
+option(ENABLE_INTERNAL_LIBJUICE "Enable internal libjuice?" ON) # TODO: Change to OFF
+option(ENABLE_INTERNAL_LIBSRTP "Enable internal libsrtp?" ON) # TODO: Change to OFF
+option(ENABLE_INTERNAL_LIBTORRENT "Enable internal libtorrent?" ON) # TODO: Change to OFF
+option(ENABLE_INTERNAL_PLOG "Enable internal plog?" ON) # TODO: Change to OFF
+option(ENABLE_INTERNAL_TRY_SIGNAL "Enable internal try_signal?" ON) # TODO: Change to OFF
+option(ENABLE_INTERNAL_USRSCTP "Enable internal usrsctp?" ON) # TODO: Change to OFF
```

Comment:
In general you wont need any of this unless its explicitly needed by libkodi. If they are dependencies of eg libtorrent, then the inclusion/search will be handled by the libtorrent find module.

The only exception i can see at a glance is boost, as you actually include a boost header in one of the changes.

Reply:
It's complicated, because we actually care about a libtorrent dependency, i.e. libdatachannel, as we want to make WebRTC connections independently.

Originally I had a single option. `ENABLE_INTERNAL_LIBTORRENT`. But then internal libjuice/libsrtp/plog/usrsctp would be dependent on another library, and you mentioned they should have their own "INTERNAL_*" variable instead of cross-depending on libtorrent.

Reply:
if you care about a dependency, put a REQUIRED on the find call, and handle the failure case.

From what i can see, libtorrent is dependent on libjuice/libsrtp/plog/usrsctp, not the other way around. The enable would be to enable libtorrent, and then in the find module for libtorrent it should handle its dependencies. If those dependences are required, then make their find modules search/or build for that lib. libkodi isnt dependendant on them however.

Reply:
If you could look into this later (post-v21, no rush) that would be appreciated. My `magnet-links` branch will remain here for collaboration. And again, post-v21, no rush!